pkg/ingester/flush.go

```diff
@@ -147,6 +148,11 @@ func (i *Ingester) flushLoop(j int) {
 		// If we're exiting & we failed to flush, put the failed operation
 		// back in the queue at a later point.
 		if op.immediate && err != nil {
+			if i.cfg.MaxFlushRetries > 0 && op.try > i.cfg.MaxFlushRetries {
+				continue
```

Comment:
We should add a log line here.

Reply:
good call

Reply:
This `continue` means that the data is lost? If yes, in addition to logs, we also need a prometheus metrics to send alarms. This metrics is used to notify our SRE to quickly check whether there is a problem with S3 and Cassandra server.

Reply:
good observation. But just fyi, the idea was to have `maxflushretries` as 0 by default to make sure you'd only suffer from it in case you're already aware of its implications.